lib/integrations/slack.rb

```diff
@@ -31,93 +29,138 @@ def send_event
-        value: "<#{payload[:frontend_url]}?expandedGroups%5B%5D=no_result | View all tests with no result>",
-        short: true
-      }
-    ]
+     {
+      fallback: message_text(fallback: true),
+      text: message_text,
+      color: "good",
+      fields: [
+        { title: "Result", value: run[:result].humanize, short: true },
+        {
+          title: "Tests Passed: #{run[:total_passed_tests]} - #{test_percentage(run[:total_passed_tests])}%",
+          value: "<#{payload[:frontend_url]}?expandedGroups%5B%5D=passed | View all Passed tests>",
+          short: true
+        },
+        { title: "Duration", value: humanize_secs(run[:time_taken]), short: true },
+        {
+          title: "Tests Failed: #{run[:total_failed_tests]} - #{test_percentage(run[:total_failed_tests])}%",
+          value: "<#{payload[:frontend_url]}?expandedGroups%5B%5D=failed | View all Failed tests>",
+          short: true
+        },
+        { title: "Environment", value: run[:environment][:name], short: true },
+        {
+          title: "Other Results: #{run[:total_no_result_tests]} - #{test_percentage(run[:total_no_result_tests])}%",
+          value: "<#{payload[:frontend_url]}?expandedGroups%5B%5D=no_result | View all tests with no result>",
+          short: true
+        }
+      ]
+    }
   end
 
   def run_error_fields
     if run[:error_reason].nil? || run[:error_reason].empty?
       run[:error_reason] = "Error reason was unspecified (please contact help@rainforestqa.com if you'd like help debugging this)"
     end
-    [{ title: "Error Reason", value: run[:error_reason], short: false }]
+    {
+      fallback: message_text(fallback: true),
+      text: message_text,
+      color: "danger",
+      fields: [{ title: "Error Reason", value: run[:error_reason], short: false }]
+    }
+    
   end
 
   def run_test_failure_fields
     failed_test = payload[:failed_test]
-    [
-      {
-        title: "Failed Test",
-        value: "<#{failed_test[:frontend_url]} | Test ##{failed_test[:id]}: #{failed_test[:title]}>",
-        short: true
-      },
-      { title: "Environment", value: run[:environment][:name], short: true },
-      { title: "Browser", value: payload[:browser][:description], short: true }
-    ]
+    {
+      fallback: message_text(fallback: true),
+      text: message_text,
+      color: "danger",
+      fields: [
+        {
+          title: "Failed Test",
+          value: "<#{failed_test[:frontend_url]} | Test ##{failed_test[:id]}: #{failed_test[:title]}>",
+          short: true
+        },
+        { title: "Environment", value: run[:environment][:name], short: true },
+        { title: "Browser", value: payload[:browser][:description], short: true }
+      ]
+    }
   end
 
   def webhook_timeout_fields
-    [{ title: "Environment", value: run[:environment][:name], short: false }]
+    {
+      fallback: message_text(fallback: true),
+      text: message_text,
```

Comment:
this is kind of a nitpick, but every method contains these two lines. is there a way to keep this DRYer?